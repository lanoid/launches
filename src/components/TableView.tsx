import * as React from 'react';

interface Props {
    launchData: any;
}

export default class TableView extends React.Component<Props> {
    constructor(props: Props){
        super(props);
    }
    
    launchRows = (launchData: any) => {
        return launchData.map((launch: any, i: number) => 
            <tr key={i}>
                <td>{launch.name}</td>
                <td>{launch.location.name}</td>
                <td>{launch.windowstart}</td>
                <td>{launch.windowend}</td>
            </tr>
        )
    }
    
    render () {
        const { launchData } = this.props;
        
        return (
            <table>
                <thead>
                    <tr>
                        <th>Name</th>                    
                        <th>Location</th>
                        <th>Launch Window Start</th>
                        <th>Launch Window End</th>
                    </tr>
                </thead>
                <tbody>
                    {this.launchRows(launchData)}
                </tbody>
            </table>
        );
    }
}