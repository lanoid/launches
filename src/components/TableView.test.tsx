import * as React from 'react';
import { shallow } from 'enzyme';
import TableView from './TableView';

describe('<TableView />', () => {
    const rocketData = [
        {
            name: 'Daft Punk',
            location: {
                name: 'My house'
            },
            windowstart: 'November 14, 2017 18:28:00 UTC',
            windowend: 'November 14, 2017 18:53:00 UTC'
        }
    ];
    const component = shallow(<TableView launchData={rocketData} />);

    it('renders rocket data accurately', () => {
        expect(component.find('td')).toHaveLength(4);
        expect(component.find('td').get(0).props.children).toBe(rocketData[0].name);
        expect(component.find('td').get(1).props.children).toBe(rocketData[0].location.name);
        expect(component.find('td').get(2).props.children).toBe(rocketData[0].windowstart);
        expect(component.find('td').get(3).props.children).toBe(rocketData[0].windowend);
    });

    it('matches the snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});