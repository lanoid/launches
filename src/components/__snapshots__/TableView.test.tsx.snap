// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableView /> matches the snapshot 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Location
      </th>
      <th>
        Launch Window Start
      </th>
      <th>
        Launch Window End
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td>
        Daft Punk
      </td>
      <td>
        My house
      </td>
      <td>
        November 14, 2017 18:28:00 UTC
      </td>
      <td>
        November 14, 2017 18:53:00 UTC
      </td>
    </tr>
  </tbody>
</table>
`;
